### PR TITLE
Feature/#169485517 support rate and time uom

### PR DESCRIPTION
**Pull request by laraferroni**
Adds Time UOM and Rate class to the unit gem so we can handle patches (and infusions at some point)

### CODE REVIEW

File: test/rate_test.rb [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+    end
+  end
+
+  context "#equivalise" do
+    [
+      ['1 mcg/hr', '1 mg/hr', '1 mcg/hr', '1000 mcg/hr'],
+      ['1 ml/hr', '1 l/hr', '1 ml/hr', '1000 ml/hr']
+    ].each do |u1, u2, expected_1, expected_2|
+      should "equivalise #{u1} and #{u2}" do
+
```
Comment (dmcblue) [inline test/rate_test.rb]: Intentionally blank?

File: Gemfile.lock [hunk 2]
```diff
@@ -0,0 +1,61 @@
+PATH
```
Comment (dmcblue) [inline Gemfile.lock]: Did we really never have the Gemfile.lock committed before? Was there a reason for that?

File: lib/unit/lexer.rb [hunk 3]
```diff
@@ -78,7 +78,10 @@ def _next_token
                   when (text = @ss.scan(/\b(?:ml|l)\b/i))
                      action { [:VOLUME_UOM, text] }
 
-                  when (text = @ss.scan(/\b(?:unit|u)\b/i))
+                  when (text = @ss.scan(/\b(?:hr)\b/i))
+                     action { [:TIME_UOM, text] }
+
+                  when (text = @ss.scan(/\b(?:unit|u|patch)\b/i))
```
Comment (doyle) [inline lib/unit/lexer.rb]: In the past we've kept medical units of measure out of this gem by converting things like caps, and tabs into unit before sending them to the units gem. That happens here: https://github.com/kitcheck/dispensecheck/blob/master/lib/dispense_check/adapter/parser_helper.rb#L122-L135

Comment (laraferroni) [reply lib/unit/lexer.rb]: Yeah, I started that way... but the issue was  that they  want it to  appear  as  "patch"  vs  "Unit" and the front end doesn't  always  have the  form  factor  to decide whether  to  show  as patch or  unit... and it felt  like  quite  a bit  of gymnastics to find all of the places that need to  show patch vs unit and get the form factor to special  case  them.  If that's the better way to go,  we can do  it, but  it  felt like this was  cleaner.

File: Gemfile.lock (hunk 2 above)
Comment (laraferroni) [reply Gemfile.lock]: no one  knew  why, but  we determined it  should be committed.

File: test/rate_test.rb (hunk 1 above)
Comment (laraferroni) [reply test/rate_test.rb]: oops.  no. this was copied  from concentration_test... it  was  blank there.  I meant to actually write a test for it though.

File: lib/unit/lexer.rb (hunk 3 above)
Comment (laraferroni) [reply lib/unit/lexer.rb]: also,  perhaps more importantly, the Unit.parse gem gets  called directly in quite a  few  places vs calling the  parser_helper method.  

https://github.com/kitcheck/dispensecheck-service/blob/6cb3b0718f57566af9efb5f3eca02d75618423a7/app/controllers/api/v1/medications_controller.rb#L136

Comment (doyle) [reply lib/unit/lexer.rb]: Ok that makes sense to me

**Pull request merged**